### DISCUSSION TITLE
Pump Rotordynamics

### DISCUSSION THREAD

**Discussion by lookid13** [petrobras/ross#935] [Q&A]
Hello,

First, thank you for this software, it is great :D.
I am interested in pump rotordynamics and follow [Corbo-1996](https://www.911metallurgist.com/blog/wp-content/uploads/2016/01/Pump-Rotordynamics-Made-Simple.pdf).

My questions regard how to model the added effects special to hydraulic pumps with regards to pneumatic turbomachinery.
Following the paper from Corbo:

1.  Impeller/Diffuser Interaction p24
The following coefficients are defined for forces generated by the impeller. The coefficients are given (eq. (70) (71)).
- Should I use a disk + seal on the same node to model the all matrices?
- How to add the forces (F0x, F0y) ?
- Should these effects be added "on top" of the "dry" disk mass?
![image](https://user-images.githubusercontent.com/27273983/209538601-5f2e623a-64d4-42e4-9024-4e9ebd17a565.png)

2.  Fluid immersion effects p26
Regarding Added mass eq (76), Stiffness reduction (78), cross-coupling (79), damping (82), mass coeff (83)
- How to add these effects to each shaft element concerned?

I am happy to provide a generic case if it helps solving the problem or once the problem is solved for future pump-struggling people :)

Thank you

**Comment by LeifParr** [comment]
Hi. 
I used to do some pump rotordynamics for multi-stage radial pumps.

As far as I know, Ross does not support mass coefficients in interconnections, e.g. the [M] and [m] in your equations above. I believe it is only stiffness and damping. 

And if I remember correctly, the cross-coupled mass-term [m] used to be small for the seals and for the impeller/diffuser interaction? So to use the current capabilities of Ross, I would suggest to model the mass [M] as a disk element, and the impeller/diffuser interaction as a bearing or seal in one node. 
Actually, the last pump I ran, the impeller/diffuser reaction coefficients were quite small compared to the effects of the annular seals. So I ended up discarding the impeller/diffusers, and using one bearing element for each seal, the front and back seal, e.g. 2 bearing elements per impeller.

I believe that the only difference between a bearing element and a seal element is how they are treated in the static analysis, e.g. the static bending line of the shaft line if it is horizontal + reaction forces. I would say this static bending line is not really of interest, since you would have to account for the seals during operation to get the correct bearing reaction forces?

You are probably already aware of this, but the book "Turbomachinery Rotordynamics: Phenomena, Modeling, and Analysis" by Dara Childs is very comprehensive, especially for pumps. I believe it has several models with values for impeller / diffuser interactions, among other things.

**Reply by lookid13** [reply]
Hello,

thank you.

I prepared a random base case based on your comments for anyone interested. It is a submerged electric motor pump. It is a jupyter notebook file in the zip joined.
[ross-pump-example.zip](https://github.com/ross-rotordynamics/ross/files/10328740/ross-pump-example.zip)
It is possible to play with 3 effects:


```py
# I miss annular seal at impeller inlet, will do
impeller_diffuser_interaction = True # Modelled with seal+disk without inertia at impeller node
liquid_filled_annuli_effect = True # Motor rotor rotating in small gap, modelled with seal+disk at center of rotor node
submerged_shaft_effects = True # added mass of shaft rotating in liquid, modelled disk without inertia at axial center of a shaft part

"""
Results:
(amplitudes are high because I let an unrealistically high hydraulic unbalance)
FFF -> CriticalSpeed=35750RPM - MaxAmplitude=11.2mm -> Dry rotor case, "base case"
TFF -> CriticalSpeed=35750RPM - MaxAmplitude=11.5mm -> Barely no effect of impeller_diffuser_interaction
FTF -> CriticalSpeed=31000RPM - MaxAmplitude=11.3mm -> -15% on critical speed with liquid_filled_annuli_effect (submerged motor)
FFT -> CriticalSpeed=35750RPM - MaxAmplitude=12.4mm -> Barely no effect of submerged_shaft_effects (added mass)
TTT -> CriticalSpeed=31000RPM - MaxAmplitude=6.0 mm -> Combined effect helped decreasing max amplitude by half
"""
```
Rotor model with all effects to `True`
![image](https://user-images.githubusercontent.com/27273983/210170075-5302f274-dd6b-4ff5-884f-4b772e1ef474.png)

To come back to your remarks:



- Important to precise: I am a beginner, I come from hydraulic design background, never really performed Rotordynamics
- cross-coupled mass-term [m] being small: It is relatively high, my only reference is from Corbo. 
![image](https://user-images.githubusercontent.com/27273983/210170150-a34337ef-167b-4df4-9d4a-9c90808f9f6b.png)
The effect from above seem to be small anyway with the "close clearance" case eq (71). I suppose it is OK to model the rotor without it
- I am not aware of the book you mentioned and could not find a digital copy online

And for some additional questions:

1. Is it OK to stack disks and seals on the same node or the n_link should be used?
2. Hydraulic Radial forces (F0x F0y) not being modeled is bothering, these forces are high. Should I add it as an "equivalent unbalance force" at the rotor speed I am interested in? Or is there a more direct way to model it?
3. In your experience, the casing stiffness should be added as a second bearing in serie or ignoring it is good enough?

**Reply by LeifParr** [reply]
Hi.
This is an interesting example, it does remind me of some pumps we worked with a while back, with a liquid-filled motor directly coupled to the pump impellers. 
I believe we spaced out the interaction between liquid and rotor along the length of the motor rotor, e.g. multiple interconnections.

For the questions, here is my take:
1. Disks and seals/interconnections on the same node is ok. n_link is used to model bearing pedestal stiffness and mass.
2. The static forces are typically used as input to obtaining reasonable bearing forces for the bearing coefficients. After that, we don't typically care about the static forces for calculating natural frequencies or forced response.
3. I assume that the casing is of a barrel type or similar. Usually, this kind of casing has a high stiffness, and it is probably good enough to ignore. Using the n_link to model pedestal stiffness assumes that each bearing has it's own path to "ground", so this may be a bit unsuited to modelling a barrel, where the bearing housings have common vibration modes.

**Reply by raphaeltimbo** [reply]
Hi!
In the near future I will add the option to add the mass terms to the bearing and seal elements.
Registered this in the following issue: #936
